### ISSUE TITLE
Delete items

### ISSUE THREAD

**Issue by gallarotti** [tkutcher/quickbase-client#56]
**Is your feature request related to a problem? Please describe.**
I may be misreading the documentation, but cannot find a simple way to delete items in Quickbase. Do I need to use the legacy API? If so, can you send me an example, please?

**Describe the solution you'd like**
It would be nice to have deletion added to the main API

**Comment by tkutcher** [comment]
Ah, I see. So right now (because of my own personal precautions/wants using the library) I have something where you need to explicitly set a `QuickbaseApiClient` to be allowed to delete objects. But it looks like I am not currently providing the same parameter through `QuickbaseTableClient`.

I will update QuickbaseTableClient so you can create them with the `allow_deletes` option.

I like leaving the option for explicitly controlling whether the python code should be allowed to make delete requests (since deleted things aren't really recoverable), but I am now thinking/considering the default should be to allow deletes but leave the option to disable them. If you/anyone has any thoughts on that let me know. 

I will leave the default as `allow_deletes=False` but you can create a client like so to create a client that can make deletes:

(I will also update the [`QuickbaseTable.client`](https://quickbase-client.readthedocs.io/en/latest/api_reference/models.html#quickbase_client.QuickbaseTable.client) factory method to forward keyword arguments...)

```py

client = QuickbaseTableClient(my_table, my_user_token, allow_deletes=True)

# OR

my_table.client(my_user_token, allow_deletes=True)
```

This will be released in 0.6.0 now

**Comment by gallarotti** [comment]
Hi Tim, sorry to bother you with this again, but would you be so kind to show a simple example of how to actually delete a record in a specific table? 

Honestly, I never found myself needing to call QuickbaseTableClient directly... 

The way I have been using your library is to have QBC generate the models for me (I run the command every 5 minutes to keep the models in sync with any changes performed by the managers in Quickbase).

Then, I use either the following code (essentially taken from the README of this repo) to **add** a new record, or I use the other code to execute a **query** (with `_and`,`_or`, etc.)

```
client = Project.client(QB_USER_TOKEN)
new_project= Project(
    ...
)
response = client.add_record(new_project)
```

How would I delete a specific record in the Project table?
